### ISSUE TITLE
[FEATURE] server_document(session_id=...)

### ISSUE THREAD

**Issue by timdiels** [bokeh/bokeh#10661]
When only accepting signed sessions, I would have to use `server_session` as `server_document` has no `session_id` parameter. I also need to pass some arguments which `server_session` cannot do (as passing args only makes sense when creating the session), so I need to wrap it in `pull_session`, but this unnecessarily pulls in the entire document.

My current workaround mimics `server_session` by adding the right header:

```python
from bokeh.embed import server_document
from bokeh.util.token import generate_session_id

def embed_bokeh(url, **kwargs):
    session_id = generate_session_id(signed=True)
    headers = {'Bokeh-Session-Id': session_id}
    return server_document(url=url, arguments=kwargs, headers=headers)
```

Please add a `session_id` parameter to `server_document` like the one of `server_session`.

**Comment by bryevdv** [comment]
I am 👎 on adding `session_id` to `server_document`. 

In the past there was a single function `autoload_server` that attempted to handle both cases:

* create a new document and session to embed
* embed an existing session

As a result the function had many overlapping, convoluted, and sometimes mutually incompatible parameters. Explaining how to use it was to users was a nightmare, and there were many, many questions and complaints  about it. The `server_document` and `server_session` functions were created specifically to tease apart the two explicit cases so they have much simpler, self-consistent parameter sets, and are easier to explain and use. This has been wildly successful, and part of that success was in clearly delineating that if you were providing an explicit  `session_id` it was because you were connecting to an existing session with that id (and `server_document` does not do that)

So, what is the fundamental ask here? I think it is:

> I want to be able to use `server_document` with servers that are configured to only use signed session ids

If that is the case, we can discuss how to make that possible without muddying the distinction above. One idea might be a different argument to `server_document` e.g. `signed_session_id=True` that does not require passing explicit session ids to `server_document`.

**Comment by timdiels** [comment]
Correct, I have no need for passing a session to `server_document` which already exists server side, I only need it to add a new signed session id to the request such that the bokeh server does not reject it with 403. What you suggest would work for me.
